### DISCUSSION TITLE
Badge Workshop build into the Site

### DISCUSSION THREAD

**Discussion by ghost** [RetroAchievements/RAWeb#720] [Ideas]
It would be nice to have a Badge Workshop on the Site right before uploading or updating a Badge to the Server. Badges are a important part of Achievements, the thing is that not everyone has the time or patience creating some Badges when it doesn't take much to create some good looking Badges, its usually just few simple steps and this is where the Badge Workshop could help saving time while improve the overall quality of Badges.

**Update:**
Added: loading Image File/s from Web (AchievementID and GameID)

**Step 1: Upload File**
Choose one or multiple Pictures/Screenshots from your Local Harddrive and pick one of the options.
- Direct Upload (Skip Workshop)
- Upload & Edit (Current Workshop Settings)
- Upload & Edit (New Workshop Settings)
- Load Image File/s from Web in Workshop (AchievementID or GameID for the whole Set)

> **Note:** Edit in Workshop will automatically be choosen when the uploaded Image File/s has a Higher or Lower (not recommended) Resolution than 64x64, Image Files with 64x64 Resolution will skip Step 2 and go directly to Step 3 unless you choose to Skip Workshop.

**Step 2: Zoom and Crop**
Here you could choose which part of the uploaded Image File will be used for the Badge.
- Zoom Image
- Crop selected part of the Image

**Step 3: Brightness, Contrast and Overlay**
This is the main component of the Badge Workshop, you could improve your uploaded Images by changing the Brightness and Contrast also adding some Overlay Effects via a Template created by the Community like Glass Effect and Borders and/or choosing a Template from your Local Harddrive and apply it.
- Brightness
- Contrast
- Overlay Effects (created by Community)
- Border Effects (created by Community)
- Using your own Template from your Local Harddrive

> **Note:** New Templates could be added anytime by the Community, we could also make a Template Contests with a voting for which Template should be added to the Workshop to keep the Quality High.

**Step 4: Repeat**
Save the current Settings and go back to Step 1 **Upload (Current Workshop Settings)**.

**Optional**
- Temporary saving the current Settings (Step 3) and applying it to all further uploads
- Batch Process for multiple Images at once and applying current Settings (Step 3) to all of them, improper Image Resolutions will be put in a Zoom and Crop schedule (Step 2).

**Comment by julkasalsa** [comment]
Great idea! Would love to have pixel accuracy options there (at least unfiltered aspect ratio x2 and x4 zoom).

**Comment by CoolCanuck** [comment]
Adobe (Avairy) is out of the question since they prohibit use in open source projects. 
https://www.adobe.io/apis/creativecloud/creativesdk.html#/creative-tools.html 

### Here's some **inspiration**:
Minecraft is the first thing that came to mind when Julia mentioned per-pixel editing. 

Minecraft Skin Editor (even works offline)
 - https://minecraft.novaskin.me/

Others:
 - https://www.minecraftskins.com/skin-editor/
- https://www.minecraftskins.net/skineditor
- https://github.com/tom-bird/pixelart  
- https://www.pixelartcss.com/    (on github: https://github.com/jvalen/pixel-art-react/)

VERY basic:
- https://www.favicon.cc/

**Comment by meleu** [comment]
@CoolCanuck yay! this one looks promising!:
https://www.pixelartcss.com/

I'm glad the guy posted the source at github :+1:

**Comment by luchaos** [comment]
I don't think Julia meant per-pixel editing but nearest-neighbour scaling within a specialized editor tool. Would you actually want to _draw_ a badge on-site?
There is use for a pixel editor for another feature i already had in mind - not necessarily for creating badges.

**Comment by SyrianBallaS** [comment]
If you like pixel editing, there's piskel.

**Comment by televandalist** [comment]
Aseprite or bust
